src/Definition/RuntimeDefinition.php

```diff
@@ -16,10 +16,10 @@
  */
 class RuntimeDefinition implements DefinitionInterface
 {
-    /** @var ClassDefinition[] */
+    /** @var array<string, ClassDefinition> */
     private $definition = [];
 
-    /** @var bool[] */
+    /** @var array<string, bool>|null */
```

Comment:
if setting an empty array as default, we could avoid having `null` here imho

Reply:
Probably yes, but this would require additional refactoring to this class since `null` is used by getter methods. I just added the types to reflect the existing behaviour.